packages/compass/src/app/index.js

```diff
@@ -69,10 +69,18 @@ window.addEventListener('error', (event) => {
 
 const darkreaderOptions = { brightness: 100, contrast: 90, sepia: 10 };
 function enableDarkTheme() {
+  // Compass-home initializes the theme and listens to these events
+  // to update the theme in the react context.
+  global.hadronApp.theme = 'DARK';
+  global.hadronApp.appRegistry?.emit('darkmode-enable');
```

Comment:
I think we could initialize the theme preference, listen to updates in the theme from the main compass process, and propagate that state to the react renderer in a more concise, generalized way which would hopefully free up adding more settings to Compass. Since the theme is the only state of Compass which is currently managed between multiple windows, I think it's alright to have the solution in this pr for now to unblock the connect form and give the theme to the react context. We can improve and generalize the initialization and synchronization of the shared state in future work when there's more to sync. What do y'all think? Should we do more now?

Reply:
I think is fine for this PR, we can do that work when we will add user settings.

One improvement I'd really love to see is to extract all of this "theme" related code from this already bloated `index.js`, including the initialization: https://github.com/mongodb-js/compass/blob/dad2a58c23b888f2b832bb6458e155b1f6ccff5c/packages/compass/src/app/index.js#L391

We did something similar for the logging: https://github.com/mongodb-js/compass/blob/ca7f24fca54d7249696821edb1417301253aad3e/packages/compass/src/main/logging.ts

Among just cleaning up that file a bit would also enable us to write some tests for it at some point